rpc/types.go

```diff
@@ -99,10 +100,21 @@ func (bn *BlockNumber) UnmarshalJSON(data []byte) error {
 		return nil
 	}
 
-	blckNum, err := hexutil.DecodeUint64(input)
+	var blckNum uint64
+	var err error
+
+	//Check if input is valid hex string before converting.
+	if hexutil.IsValidHexString(input) {
+		blckNum, err = hexutil.DecodeUint64(input)
+	} else {
+		//Else try converting input directly into uint64 value
+		blckNum, err = strconv.ParseUint(input, 10, 64)
+	}
+
```

Comment:
```go
blckNum, err := hexutil.DecodeUint64(input)
if err != nil {
    blckNum, err = strconv.ParseUint(input, 10, 64)
}
```

Reply:
Hi @gzliudan ,

I have added a new public method `IsValidHexString` which has more comprehensive check for hex string validity. `IsValidHexString` internally also using `has0xPrefix` with other checks like valid character in string.

Reply:
It is not a good idea that check with your `IsValidHexString`. eg: when input is "0xYZ", `hexutil.IsValidHexString(input)` will returns false, then `strconv.ParseUint(input, 10, 64)` will fail to execute, and the final fail reason is not true for "0xYZ". In fact, if `hexutil.has0xPrefix(input)` is ture, maybe it is not a valid hex string, but it can not be a decimal integer at this time, `hexutil.DecodeUint64(input)` will tell the real fail reason.

Reply:
Got it, thanks for insight @gzliudan, I am making the required changes.